0011-interactive-transport-protocol/0011-interactive-transport-protocol.md

```diff
@@ -0,0 +1,104 @@
+     ------------------
+
+The [Simple Payment Setup Protocol (SPSP)](../0009-simple-payment-setup-protocol) uses the Interactive Transport Protocol (ITP). ITP is a host-to-host protocol intended for use on top of the Interledger Protocol.
+
+### Model of Operation
+
+We suppose that a sender wants to pay a receiver for some good or service and that the two are using a communication channel specified by a higher level protocol.
+
+We suppose that the receiver has a 32-byte secret value.
+
+1. The receiver creates an Interledger Packet using the details of the payment they want to request from the sender. The `condition` field is left undefined.
+2. The receiver creates the ITP Header (specified [below](#specification)) using a [UUID (version 4 is RECOMMENDED)](https://www.ietf.org/rfc/rfc4122.txt) and a token or string associated with the sender and/or good being purchased.
+3. The receiver attaches the ITP header to the Interledger Packet.
+4. The receiver encodes the complete Interledger Packet in the [Canonical OER](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview%20of%20OER.pdf) format.
+5. The receiver creates an [HMAC](https://tools.ietf.org/html/rfc2104) of the encoded packet using the SHA-256 hash function and the receiver's secret as the HMAC key.
+6. The receiver uses the HMAC output as the _preimage_ for a [PREIMAGE-SHA-256 Crypto Condition](../0002-crypto-conditions) (the HMAC output is hashed again using SHA-256 to create the condition).
+7. The receiver uses this condition as the `condition` field in the Interledger Packet.
+8. The receiver communicates the complete Interledger Packet to the sender using the communication channel.
+9. The sender inspects the packet details.
+10. The sender requests a quote for the payment from a connector.
+11. The sender sends a transfer to a connector with the packet from the recipient attached.
+12. Connectors forward the packet until it reaches the receiver's ledger and the receiver is notified of funds on hold.
+13. The receiver checks the transfer against the packet details. The receiver checks, among other things, that the transfer amount is greater than or equal to the amount specified in the packet and that the packet is not yet expired. If any of the checks fail, the receiver rejects the transfer or lets the hold expire and skips all further steps.
+14. The receiver **regenerates** the condition fulfillment, following steps 4-6.
+15. The receiver checks that the condition generated matches the one included in the incoming transfer. If it does not, the receiver rejects the transfer or lets the hold expire and skips all further steps.
+16. The receiver submits the condition fulfillment to their ledger to claim the funds.
+17. When the receiver's ledger confirms that the fulfillment has been received and the funds transferred, the receiver uses the `requestUniqueId` and the `receiverRequestPurpose` fields from the ITP Header to credit the sender accordingly.
+
```

Comment:
Instead of saying the sender is credited I'd say that the receiver uses these fields to reconcile the incoming transfer with the original payment request